### PR TITLE
`Notification` validation

### PR DESCRIPTION
**Pull request by 4view**

### CODE REVIEW

File: src/Notificator.Infrastructure/Validation/NotificationValidationService.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            throw new InvalidOperationException("Notification StartTime must be greater then current time");
+        if (notificationDto.MaxAmount < 0)
+            throw new InvalidOperationException("Notificator MaxAmount can`t be less then 0");
+        if (notificationDto.Interval != null || notificationDto.Interval < _settings.MinInterval)
+            throw new InvalidOperationException("Notificator Interval could be empty or Interval value must be greater then MinInterval");
+        if (notificationDto.EndTime != null ||  notificationDto.EndTime < notificationDto.StartTime && (notificationDto.StartTime - notificationDto.EndTime) < _settings.MinInterval)
+            throw new InvalidOperationException("Notificator EndTime could be empty or he can`t be less then StartTime "
+            + "and difference between StartTime and EndTime shouldn't be less then MinInterval");
+        
+
```
Comment (CIRCULARKA) [inline src/Notificator.Infrastructure/Validation/NotificationValidationService.cs:37]: Два лишних пробела? Ты сумасшедший

File: src/Notificator.Infrastructure/Validation/NotificationValidationService.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        _timeService = timeService;
+    }
+
+    /// <summary>
+    /// Validates notification. Takes into account validation settings
+    /// </summary>
+    public void Validation(NotificationDto notificationDto) 
+    {
+        if (string.IsNullOrWhiteSpace(notificationDto.Text))
+            throw new InvalidOperationException("Notification Text must not be emtpy");
```
Comment (CIRCULARKA) [inline src/Notificator.Infrastructure/Validation/NotificationValidationService.cs:24]: "Notification's text must not be empty"

File: src/Notificator.Infrastructure/Validation/NotificationValidationService.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        if (string.IsNullOrWhiteSpace(notificationDto.Header) && _settings.IsHeaderRequired)
+            throw new InvalidOperationException("Notification Header must not be empty");
```
Comment (CIRCULARKA) [inline src/Notificator.Infrastructure/Validation/NotificationValidationService.cs:26]: "Notification's header must not be empty"

File: src/Notificator.Infrastructure/Validation/NotificationValidationService.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        if (notificationDto.StartTime > _timeService.CurrentTime)
+            throw new InvalidOperationException("Notification StartTime must be greater then current time");
```
Comment (CIRCULARKA) [inline src/Notificator.Infrastructure/Validation/NotificationValidationService.cs:28]: "Notification's start time must be greater than the current time"

File: src/Notificator.Infrastructure/Validation/NotificationValidationService.cs [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        if (notificationDto.MaxAmount < 0)
+            throw new InvalidOperationException("Notificator MaxAmount can`t be less then 0");
```
Comment (CIRCULARKA) [inline src/Notificator.Infrastructure/Validation/NotificationValidationService.cs:30]: "Max amount of notifications must be greater than zero"

File: src/Notificator.Infrastructure/Validation/NotificationValidationService.cs [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        if (notificationDto.Interval != null || notificationDto.Interval < _settings.MinInterval)
+            throw new InvalidOperationException("Notificator Interval could be empty or Interval value must be greater then MinInterval");
```
Comment (CIRCULARKA) [inline src/Notificator.Infrastructure/Validation/NotificationValidationService.cs:32]: Кто такой этот ваш нотификатор. $"Notification's interval must not be less than the minimal interval of {_settings.MinInterval}"

File: src/Notificator.Infrastructure/Validation/NotificationValidationService.cs [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+        if (notificationDto.EndTime != null ||  notificationDto.EndTime < notificationDto.StartTime && (notificationDto.StartTime - notificationDto.EndTime) < _settings.MinInterval)
+            throw new InvalidOperationException("Notificator EndTime could be empty or he can`t be less then StartTime "
+            + "and difference between StartTime and EndTime shouldn't be less then MinInterval");
```
Comment (CIRCULARKA) [inline src/Notificator.Infrastructure/Validation/NotificationValidationService.cs:35]: Слишком сложное, большое условие. Нужно разбить его на три последовательных проверки

**Review by CIRCULARKA — CHANGES_REQUESTED** [review]

File: src/Notificator.Infrastructure/Validation/NotificationValidationService.cs (hunk 3 above)
Comment (4view) [reply src/Notificator.Infrastructure/Validation/NotificationValidationService.cs]: @CIRCULARKA Fixed https://github.com/CIRCULARKA/notificator/pull/10/commits/03be5ba7c13d4119bb23f27620ce125dedfb3bd1

File: src/Notificator.Infrastructure/Validation/NotificationValidationService.cs (hunk 4 above)
Comment (4view) [reply src/Notificator.Infrastructure/Validation/NotificationValidationService.cs]: @CIRCULARKA Fixed https://github.com/CIRCULARKA/notificator/commit/03be5ba7c13d4119bb23f27620ce125dedfb3bd1

File: src/Notificator.Infrastructure/Validation/NotificationValidationService.cs (hunk 5 above)
Comment (4view) [reply src/Notificator.Infrastructure/Validation/NotificationValidationService.cs]: @CIRCULARKA Fixed https://github.com/CIRCULARKA/notificator/commit/03be5ba7c13d4119bb23f27620ce125dedfb3bd1

File: src/Notificator.Infrastructure/Validation/NotificationValidationService.cs (hunk 6 above)
Comment (4view) [reply src/Notificator.Infrastructure/Validation/NotificationValidationService.cs]: @CIRCULARKA Fixed https://github.com/CIRCULARKA/notificator/commit/03be5ba7c13d4119bb23f27620ce125dedfb3bd1

File: src/Notificator.Infrastructure/Validation/NotificationValidationService.cs (hunk 7 above)
Comment (4view) [reply src/Notificator.Infrastructure/Validation/NotificationValidationService.cs]: @CIRCULARKA Fixed https://github.com/CIRCULARKA/notificator/commit/03be5ba7c13d4119bb23f27620ce125dedfb3bd1

File: src/Notificator.Infrastructure/Validation/NotificationValidationService.cs [hunk 8]
```diff
@@ -34,5 +34,14 @@ public void Validation(NotificationDto notificationDto)
...
         if ((notificationDto.StartTime - notificationDto.EndTime) < _settings.MinInterval)
             throw new InvalidOperationException("Notification's {_settings.MinInterval} must be greater then difference between start time and end time");
+        if (notificationDto.DaysOfTheWeek == null)
+            return;
+        notificationDto.DaysOfTheWeek.ForEach(day => 
+        {
+            if (day < 1 || day > 7)
+                throw new InvalidOperationException("Day must be more then zero and less then seven");
+        });
+        if (notificationDto.DaysOfTheWeek.Distinct().Count() != notificationDto.DaysOfTheWeek.Count)
```
[the remark this answers is not in the document]
Comment (4view) [reply src/Notificator.Infrastructure/Validation/NotificationValidationService.cs]: @CIRCULARKA fixed  https://github.com/CIRCULARKA/notificator/pull/10/commits/d4fbb6d36223e7956e043cd857c27139607929c7

**Review by CIRCULARKA — APPROVED** [review]

**Pull request merged**